src/main/scala/io/iohk/ethereum/consensus/Consensus.scala

```diff
@@ -0,0 +1,41 @@
+package io.iohk.ethereum.consensus
+
+import io.iohk.ethereum.domain.Block
+import io.iohk.ethereum.ledger.BlockExecutionError.ValidationBeforeExecError
+import io.iohk.ethereum.ledger.BlockExecutionSuccess
+import io.iohk.ethereum.nodebuilder.Node
+import io.iohk.ethereum.validators.BlockHeaderValidator
+
+/**
+ * Abstraction for a consensus protocol.
+ *
+ * @see [[io.iohk.ethereum.consensus.Protocol Protocol]]
+ */
+// FIXME Lot's of stuff to do...
+trait Consensus {
+  type Config <: AnyRef /*Product*/
+
+  def protocol: Protocol
+
+  def config: FullConsensusConfig[Config]
+
+  /** Returns `true` if this is the standard Ethereum PoW consensus protocol (`ethash`). */
+  final def isEthash: Boolean = protocol.isEthash
```

Comment:
I didn't find any usages for it, which is good because I don't think such methods belong in object oriented design.

Reply:
Here is the story of `isEthash` and `ifEthash`. Some code in EthService would need to run conditionally only if we are running under Ethash consensus. Also there would be a sanity check in a test case that Mantis includes at least Ethash as consensus. Thus the introduction of `isEthash`. Of course, `ifEthash` subsumes this functionality and provides an even more general one, whether more exact details are provided as a parameter. This means we can retire `isEthash`. And regarding `ifEthash`, essentially a polymorphic `fold` over ethash and non-ethash consensus, I was thinking of moving it to a utility class (an `implicit class extends AnyVal`), so as the Consensus interface is clean of such functionality. I suggest we revisit these design decisions in the next PR.

Reply:
What do you propose we do to remember we need to revisit? A `FIXME` comment, or a YT ticket?

Reply:
I am looking at it almost every day while working on the next PR, so it will not be forgotten. Most probably the resolution is going to look like this: Since we absolutely need this polymorphic decision making the `ifEthash` is going to be implemented somewhere, and `isEthash` is not fundamental: it is a utility easily implemented on top of `ifEthash`. Most probably, I will get `ifEthash` out of `Consensus` as an extention method (via the `implicit class extends Any` mechanism).